src/domain/core/src/services/export_service.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::Path;
+
+use datafusion::dataframe::DataFrame;
+use internal_error::{BoxedError, InternalError};
+use thiserror::Error;
+
+use crate::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Service
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait ExportService: Send + Sync {
+    async fn export_to_fs(
+        &self,
+        df: DataFrame,
+        path: &Path,
+        format: &ExportFormat,
+        records_per_file: Option<usize>,
```

Comment:
Let's add `ExportOpts` struct and move this parameter there. This way users will be able to pass `ExportOpts::default()` where they don't care about the details, and we will be able to add more options without breaking every place where this method is called.

This is a pattern a lot of our services follow.